AdyenComponents/BACS Direct Debit/BACSDirectDebitComponent.swift

```diff
@@ -0,0 +1,101 @@
+//
+// Copyright (c) 2021 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import Adyen
+import UIKit
+
+internal protocol BACSDirectDebitRouterProtocol {
+    func presentConfirmation(with data: BACSDirectDebitData)
+    func confirmPayment(with data: BACSDirectDebitData)
+    func cancelPayment()
+}
+
+/// A component that provides a form for BACS Direct Debit payments.
+public final class BACSDirectDebitComponent: PaymentComponent, PresentableComponent {
+
+    // MARK: - PresentableComponent
+
+    public var viewController: UIViewController
+
+    public var requiresModalPresentation: Bool = false
+
+    /// The object that acts as the delegate of the component.
+    public weak var delegate: PaymentComponentDelegate?
+
+    /// The BACS Direct Debit payment method.
+    public let paymentMethod: PaymentMethod
+
+    /// :nodoc:
+    public let apiContext: APIContext
+
+    /// :nodoc:
+    public let style: FormComponentStyle
+
+    /// :nodoc:
+    public var localizationParameters: LocalizationParameters?
+
+    // MARK: - Initializers
+
+    /// Creates and returns a BACS Direct Debit component.
+    /// - Parameters:
+    ///   - paymentMethod: The BACS Direct Debit payment method.
+    ///   - apiContext: The API context.
+    ///   - style: The component's UI style.
+    ///   - localizationParameters: The localization parameters.
+    public init(paymentMethod: BACSDirectDebitPaymentMethod,
+                apiContext: APIContext,
+                style: FormComponentStyle = .init(),
+                localizationParameters: LocalizationParameters? = nil) {
+        self.paymentMethod = paymentMethod
+        self.apiContext = apiContext
+        self.style = style
+        self.localizationParameters = localizationParameters
+
+        let view = BACSDirectDebitInputFormViewController(title: paymentMethod.name,
+                                                          styleProvider: style)
+        self.viewController = view as UIViewController
+
+        let itemsFactory = BACSDirectDebitItemsFactory(styleProvider: style,
+                                                       localizationParameters: localizationParameters,
+                                                       scope: self)
+        let presenter = BACSDirectDebitPresenter(view: view,
+                                                 router: self,
+                                                 itemsFactory: itemsFactory)
+        view.presenter = presenter
+    }
+}
+
+// MARK: - BACSDirectDebitRouterProtocol
+
+extension BACSDirectDebitComponent: BACSDirectDebitRouterProtocol {
```

Comment:
is this extension `internal`? can you add `internal` to its definition?

Reply:
The `BACSDirectDebitRouterProtocol` is declared with access level `internal` already. So any conformance to it will be at that access level. Also, I believe it's not possible to declare `internal` extensions that declare protocol conformances.

Reply:
Marked functions as `internal` instead.